flink-end-to-end-tests/flink-plugins-test/src/main/java/org/apache/flink/fs/dummy/DummyFs.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.fs.dummy;
+
+import org.apache.flink.core.fs.BlockLocation;
+import org.apache.flink.core.fs.FSDataInputStream;
+import org.apache.flink.core.fs.FSDataOutputStream;
+import org.apache.flink.core.fs.FileStatus;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.FileSystemKind;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.core.fs.local.LocalBlockLocation;
+
+import javax.annotation.Nullable;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
```

Comment:
Should we add a java doc with a reference to `DummyFileSystemFactory`?

Reply:
There is no clear convention. I follow up a rule, that I place a comment if something is not obvious based on the class name/line of code. Here `DummyFileSystem` doesn't explain in what why is it "dummy", so it needs some explanation. The problem is that a developer can encounter this `DummyFileSystem` either through `FileSystem` or `FileSystemFactory`, so I would place an explanation in one of them, and reference it in the other.

It hardly matters if class is a test class, or not.

Reply:
There is a description in the factory class.

Reply:
Yes, but:
> The problem is that a developer can encounter this DummyFileSystem either through FileSystem or FileSystemFactory, so I would place an explanation in one of them, and reference it in the other.

I mean it's not obvious that if you wonder what `DummyFileSystem` is doing, that you should check a comment in `DummyFileSystemFactory` class.